sycl/include/CL/sycl/buffer.hpp

```diff
@@ -54,14 +54,14 @@ class buffer {
          const property_list &propList = {})
       : Range(bufferRange), MemRange(bufferRange) {
     impl = std::make_shared<detail::buffer_impl<AllocatorT>>(
-        hostData, get_count() * sizeof(T), propList);
+        hostData, sizeof(T), get_count() * sizeof(T), propList);
```

Comment:
I'm not sure that it's the correct way. I can image a situation, where T is user defined structure with sizeof(T) is uneven number. I think it's better to implement a function roundToClosestPowerOfTwo() number and call it here.

Reply:
Good point, done.